### PR TITLE
fix strange crash with conditionally-visible references

### PR DESCRIPTION
**Pull request by andrewroth**

### CODE REVIEW

File: app/controllers/fe/concerns/answer_pages_controller_concern.rb [hunk 1]
```diff
@@ -45,7 +45,12 @@ def update
       params[:reference].keys.each do |id|
         reference_params = params.fetch(:reference)[id].permit(:relationship, :title, :first_name, :last_name, :phone, :email, :is_staff)
 
+        # The call to Application#reference_sheets is supposed to update the visibility cache and return the right references, but
+        # currently it does not do that right.  I have to call @answer_sheet.reference_sheets manually once, then it gets the right
+        # list.  I think this may be caused from a change in rails 5
+        @answer_sheet.reference_sheets
```
Comment (knutsenm) [inline app/controllers/fe/concerns/answer_pages_controller_concern.rb]: Would `@answer_sheet.reload` work?

Comment (andrewroth) [reply app/controllers/fe/concerns/answer_pages_controller_concern.rb]: Perhaps.. I'll try it

Comment (andrewroth) [reply app/controllers/fe/concerns/answer_pages_controller_concern.rb]: Yes, .reload works.  Do you think I should keep the comment explanation?

Comment (knutsenm) [reply app/controllers/fe/concerns/answer_pages_controller_concern.rb]: Yes

**Pull request merged**